### DISCUSSION TITLE
sqlalchemy disappears "from clasuse" by to use with_entities

### DISCUSSION THREAD

**Discussion by sasano8** [sqlalchemy/sqlalchemy#6041] [Usage Questions] [answered]
- Ubuntu 18.04.5 LTS
- python3.8
- SQLAlchemy = "^1.3.22"

``` python
str(db.query(User).filter_by(id=1).with_entities("id"))
# => SELECT id FROM users WHERE users.id = %(id_1)s

str(db.query(User).with_entities("id"))
# => SELECT id
```

As in the example above, the "from clause" disappears and a bad query is generated.
I don't know if it's a bug or a spec, but I'll report it.

**Comment by zzzeek** [comment, marked as answer]
hey there, I believe you intend to do this:

    db.query(User).with_entities(User.id)

"id" is just an arbitrary string without any context.  the behavior is expected.

**Comment by sasano8** [comment]
I wanted to do something like this.

```
from pydantic import BaseModel
from sqlalchemy.orm import load_only

class UserOut(BaseModel):
  id: int
  name: str

db.query(User).options(load_only(*UserOut.__fields__))
```

When loading a large amount of data, I was looking for a way to limit the columns to output.
It may be useful to be able to specify the field as a string.
And I found "load_only".(I was more happy if it to be able to use it in a method chain.)

Thanks for great liblary.

**Reply by zzzeek** [reply]
load_only() accepts string names, they are matched up to the attribute names on the "lead" entity, which is the first entity requested, in this case User.     while it might seem similar this is actually a different use case than what with_entities() does.  with_entities() fully replaces whatever it is that the query is SELECTing in the first place.

**Reply by sasano8** [reply]
Yeah, load_only perfectly matches my request.
I can't imagine a situation that uses with_entities at the moment.
